soc/arm/atmel_sam0/common/soc_samd2x.c

```diff
@@ -9,168 +10,269 @@
  * @brief Atmel SAMD MCU series initialization code
  */
 
+/* The CPU clock will be configured to the DT requested value,
+ * and run via DFLL48M.
+ *
+ * Reference -> GCLK Gen 1 -> DFLL48M -> GCLK Gen 0 -> GCLK_MAIN
+ *
+ * GCLK Gen 0 -> GCLK_MAIN
+ * GCLK Gen 1 -> DFLL48M (variable)
+ * GCLK Gen 2 -> WDT @ 32768 Hz
+ * GCLK Gen 3 -> ADC @ 8 MHz
+ */
+
 #include <zephyr/arch/cpu.h>
 #include <zephyr/arch/arm/aarch32/cortex_m/cmsis.h>
 #include <zephyr/device.h>
 #include <zephyr/init.h>
 #include <zephyr/kernel.h>
-#include <soc.h>
 
-static void flash_waitstates_init(void)
-{
-	/* One wait state at 48 MHz. */
-	NVMCTRL->CTRLB.bit.RWS = NVMCTRL_CTRLB_RWS_HALF_Val;
-}
+#include <soc.h>
 
-/* Follows the same structure as the Arduino Zero, namely:
- *  XOSC32K -> GCLK1 -> DFLL48M -> GCLK0
- *  OSC8M -> 8 MHz -> GCLK3
+/**
+ * Fix different naming conventions for SAMD20
  */
+#ifdef FUSES_OSC32KCAL_ADDR
+#define FUSES_OSC32K_CAL_ADDR		FUSES_OSC32KCAL_ADDR
+#define FUSES_OSC32K_CAL_Pos		FUSES_OSC32KCAL_Pos
+#define FUSES_OSC32K_CAL_Msk		FUSES_OSC32KCAL_Msk
+#endif
 
-static void xosc_init(void)
+#if !CONFIG_SOC_ATMEL_SAMD_OSC8M || CONFIG_SOC_ATMEL_SAMD_DEFAULT_AS_MAIN
+#define osc8m_init()
+#else
+static inline void osc8m_init(void)
 {
-#ifdef CONFIG_SOC_ATMEL_SAMD_XOSC
-#error External oscillator support is not implemented.
-#endif
+	uint32_t reg;
+
+	/* Save calibration */
+	reg = SYSCTRL->OSC8M.reg
+		& (SYSCTRL_OSC8M_FRANGE_Msk | SYSCTRL_OSC8M_CALIB_Msk);
+
+	SYSCTRL->OSC8M.reg = reg
+		| !SYSCTRL_OSC8M_ONDEMAND
+		| SYSCTRL_OSC8M_RUNSTDBY
+		| SYSCTRL_OSC8M_PRESC(0) /* 8MHz (/1) */
+		| SYSCTRL_OSC8M_ENABLE;
+
+	while (!SYSCTRL->PCLKSR.bit.OSC8MRDY) {
+	}
```

Comment:
You should keep correctly vertical alignment for operators, see below:
```diff
	uint32_t reg;

	/* Save calibration */
	reg = SYSCTRL->OSC8M.reg
	    & (SYSCTRL_OSC8M_FRANGE_Msk | SYSCTRL_OSC8M_CALIB_Msk);

	SYSCTRL->OSC8M.reg = reg
			   | !SYSCTRL_OSC8M_ONDEMAND
			   | SYSCTRL_OSC8M_RUNSTDBY
			   | SYSCTRL_OSC8M_PRESC(0) /* 8MHz (/1) */
			   | SYSCTRL_OSC8M_ENABLE;

	while (!SYSCTRL->PCLKSR.bit.OSC8MRDY) {
	}
```

Fix all occurrences.

BTW, what you want achieve with below entry?
```diff
			   | !SYSCTRL_OSC8M_ONDEMAND
```

Reply:
@nandojve 
Vertical alignment, including those unfortunate `| !SYSCTRL_OSC8M_ONDEMAND` are copy-pasted from the existing `soc_saml2x.c` implementation (thought it would be a good idea to keep the style).
The current alignment is done at a specific number of tabs. putting all `|` under `=` would require adding extra spaces and screw up with the vertical scrolling, as everything will be aligned depending on the lval expression length.

Reply:
@nandojve

Manged to find some time and align those fields. Please have a look.
I don't know if this change really helps with the visuals. They look OK only when the tab size is set to 4.
Also removed the `!ONDEMAND` stuff and the `= 0` values.

Regards,
Ionut

Reply:
You should adjust code for tab size of 8.
The rule is use tab most you can and then complement with spaces.